File: common/src/main/java/com/jmsoftware/maf/common/domain/ValidationTestPayload.java
package com.jmsoftware.maf.common.domain;

import lombok.Data;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

/**
 * <h1>ValidationTestPayload</h1>
 * <p>
 * Change description here.
 *
 * @author Johnny Miller (锺俊), email: johnnysviva@outlook.com
 * @date 2/14/20 11:34 AM
 **/
@Data
public class ValidationTestPayload {
    @NotNull
    @Min(value = 1L)
    private Long id;
    @NotBlank
    private String name;
}

File: common/src/main/kotlin/com/jmsoftware/maf/common/domain/ValidationTestPayload.kt
package com.jmsoftware.maf.common.domain

import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

/**
 * # ValidationTestPayload
 *
 * Change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/18/22 9:18 PM
 */
class ValidationTestPayload {
    @NotNull
    @Min(value = 1L)
    var id: Long = 0L

    @NotBlank
    lateinit var name: String
}
